File: docs-examples/example-python/src/main/python/micronaut/configuration/openapi/docs/generics/MyController.py
from typing import Annotated

from micronaut.http.annotation import Body, Controller, Put
from micronaut.configuration.openapi.docs.schemas.Pet import Pet

from .Response import Response


# tag::clazz[]
@Controller
class MyController:

    @Put("/")
    def updatePet(self, pet: Annotated[Pet, Body]) -> Response[Pet]:
        return Response(pet)
# end::clazz[]

File: docs-examples/example-kotlin/src/main/kotlin/io/micronaut/configuration/openapi/docs/generics/MyController.kt
package io.micronaut.configuration.openapi.docs.generics

import io.micronaut.configuration.openapi.docs.schemas.Pet
import io.micronaut.http.annotation.Controller
import io.micronaut.http.annotation.Put

// tag::clazz[]
@Controller
class MyController {

    @Put("/")
    fun updatePet(pet: Pet): Response<Pet> {
        return Response(pet)
    }
}
// end::clazz[]
